subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/DependencyGraphNode.java

```diff
@@ -40,7 +41,7 @@
 
     Set<? extends DependencyGraphEdge> getIncomingEdges();
 
-    Set<? extends DependencyGraphEdge> getOutgoingEdges();
+    Collection<? extends DependencyGraphEdge> getOutgoingEdges();
```

Comment:
Because a `Set` wasn't necessary and involved computing `hashCode()`/`equals` too often. Moving this to a simple `Collection` allows us to use an optimized structure under the hood (see commit message for details).